__tests__/distributors/zulu-installer.test.ts

```diff
@@ -0,0 +1,369 @@
+      async (url): Promise<ifm.ITypedResponse<any>> => {
+        const result = JSON.stringify(manifestData);
+        return {
+          statusCode: 200,
+          headers: {},
+          result: [
+            {
+              url: url,
+              jdk_version: []
+            }
+          ]
+        };
+      }
+    );
+
+    zuluDistributor = new ZuluDistributor(input);
+    const buildUrl = `https://api.azul.com/zulu/download/community/v1.0/bundles/${parsedUrl}`;
+    const releaseCheck = await zuluDistributor['getAvailableVersions']();
+    expect(releaseCheck[0].url).toEqual(buildUrl);
+  });
+
+  it('load available versions', async () => {
+    zuluDistributor = new ZuluDistributor({ version: '11', arch: 'x86', packageType: 'jdk' });
+    await expect(zuluDistributor['getAvailableVersions']()).resolves.not.toBeNull();
+  });
+
+  it('Error is thrown for empty response', async () => {
+    spyHttpClient.mockImplementationOnce(
+      async (): Promise<ifm.ITypedResponse<IZuluVersions[]>> => {
+        return {
+          statusCode: 200,
+          headers: {},
+          result: [] as IZuluVersions[]
+        };
+      }
+    );
+    zuluDistributor = new ZuluDistributor({ version: '11', arch: 'x86', packageType: 'jdk' });
+    await expect(zuluDistributor['getAvailableVersions']()).rejects.toThrowError(
+      /No versions were found using url */
+    );
+  });
+
+  it('Error is thrown for undefined', async () => {
+    spyHttpClient.mockImplementationOnce(
+      async (): Promise<ifm.ITypedResponse<IZuluVersions[] | undefined>> => {
+        return {
+          statusCode: 200,
+          headers: {},
+          result: undefined
+        };
+      }
+    );
+    zuluDistributor = new ZuluDistributor({ version: '11', arch: 'x86', packageType: 'jdk' });
+    await expect(zuluDistributor['getAvailableVersions']()).rejects.toThrowError(
+      /No versions were found using url */
+    );
+  });
+});
+
+describe('getArchitectureOptions', () => {
+  it.each([
+    [{ architecture: 'x64' }, { arch: 'x86', hw_bitness: '64', abi: '' }],
+    [{ architecture: 'x86' }, { arch: 'x86', hw_bitness: '32', abi: '' }],
+    [{ architecture: 'x32' }, { arch: 'x32', hw_bitness: '', abi: '' }],
+    [{ architecture: 'arm' }, { arch: 'arm', hw_bitness: '', abi: '' }]
+  ])('%o -> %o', (input, expected) => {
+    let zuluDistributor = new ZuluDistributor({
+      version: '11',
+      arch: input.architecture,
+      packageType: 'jdk'
+    });
+    expect(zuluDistributor['getArchitectureOptions']()).toEqual(expected);
+  });
+});
+
+describe('findPackageForDownload', () => {
+  let spyHttpClient: jest.SpyInstance;
+  let zuluDistributor: ZuluDistributor;
+
+  beforeEach(() => {
+    spyHttpClient = jest.spyOn(HttpClient.prototype, 'getJson');
+    spyHttpClient.mockImplementation(
+      async (): Promise<ifm.ITypedResponse<any[]>> => {
+        manifestData = require('../data/zulu/zulu-releases-default.json');
```

Comment:
Why do we need to import this file in every `beforeEach` iteration? Can we move it to the top?

Reply:
Removed. Thank you.